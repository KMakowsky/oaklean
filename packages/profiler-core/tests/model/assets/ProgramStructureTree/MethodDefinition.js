class Parent {
  constructor() {
    () => {}
    () => {}
    () => {}
  }

  methodToInherit() {

  }

  1() {

  }
}


class Child extends Parent {
  constructor() {

  }

  arrowFunction = () => {

  }

  ['ex' + 'pression' + 'arrowFunction'] = () => {

  }

  async asyncMethod() {

  }

  static staticMethod() {

  }

  get fullName() {

  }

  set fullName(value) {

  }

  method() {

  }

  [(() => { return 'dynamicMethodName' + (new Date()).getSeconds()})()]() {

  }

  [`literal`]() {

  }
}