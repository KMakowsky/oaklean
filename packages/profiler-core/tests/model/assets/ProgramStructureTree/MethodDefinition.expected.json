{
	"id": 0,
	"type": "Root",
	"identifierType": "Name",
	"identifier": "{root}",
	"beginLoc": {
		"line": 1,
		"column": 0
	},
	"endLoc": {
		"line": 58,
		"column": 1
	},
	"children": {
		"{class:Parent}": {
			"id": 1,
			"type": "ClassDeclaration",
			"identifierType": "Name",
			"identifier": "{class:Parent}",
			"beginLoc": {
				"line": 1,
				"column": 0
			},
			"endLoc": {
				"line": 15,
				"column": 1
			},
			"children": {
				"{constructor:constructor}": {
					"id": 2,
					"type": "ConstructorDeclaration",
					"identifierType": "Name",
					"identifier": "{constructor:constructor}",
					"beginLoc": {
						"line": 2,
						"column": 2
					},
					"endLoc": {
						"line": 6,
						"column": 3
					},
					"children": {
						"{functionExpression:(anonymous:0)}": {
							"id": 3,
							"type": "ArrowFunctionExpression",
							"identifierType": "Anonymous",
							"identifier": "{functionExpression:(anonymous:0)}",
							"beginLoc": {
								"line": 3,
								"column": 4
							},
							"endLoc": {
								"line": 3,
								"column": 12
							},
							"children": {}
						},
						"{functionExpression:(anonymous:1)}": {
							"id": 4,
							"type": "ArrowFunctionExpression",
							"identifierType": "Anonymous",
							"identifier": "{functionExpression:(anonymous:1)}",
							"beginLoc": {
								"line": 4,
								"column": 4
							},
							"endLoc": {
								"line": 4,
								"column": 12
							},
							"children": {}
						},
						"{functionExpression:(anonymous:2)}": {
							"id": 5,
							"type": "ArrowFunctionExpression",
							"identifierType": "Anonymous",
							"identifier": "{functionExpression:(anonymous:2)}",
							"beginLoc": {
								"line": 5,
								"column": 4
							},
							"endLoc": {
								"line": 5,
								"column": 12
							},
							"children": {}
						}
					}
				},
				"{method:methodToInherit}": {
					"id": 6,
					"type": "MethodDefinition",
					"identifierType": "Name",
					"identifier": "{method:methodToInherit}",
					"beginLoc": {
						"line": 8,
						"column": 2
					},
					"endLoc": {
						"line": 10,
						"column": 3
					},
					"children": {}
				},
				"{method:(literal:0)}": {
					"id": 7,
					"type": "MethodDefinition",
					"identifierType": "Literal",
					"identifier": "{method:(literal:0)}",
					"beginLoc": {
						"line": 12,
						"column": 2
					},
					"endLoc": {
						"line": 14,
						"column": 3
					},
					"children": {}
				}
			}
		},
		"{class:Child}": {
			"id": 8,
			"type": "ClassDeclaration",
			"identifierType": "Name",
			"identifier": "{class:Child}",
			"beginLoc": {
				"line": 18,
				"column": 0
			},
			"endLoc": {
				"line": 58,
				"column": 1
			},
			"children": {
				"{constructor:constructor}": {
					"id": 9,
					"type": "ConstructorDeclaration",
					"identifierType": "Name",
					"identifier": "{constructor:constructor}",
					"beginLoc": {
						"line": 19,
						"column": 2
					},
					"endLoc": {
						"line": 21,
						"column": 3
					},
					"children": {}
				},
				"{functionExpression:arrowFunction}": {
					"id": 10,
					"type": "ArrowFunctionExpression",
					"identifierType": "Name",
					"identifier": "{functionExpression:arrowFunction}",
					"beginLoc": {
						"line": 23,
						"column": 18
					},
					"endLoc": {
						"line": 25,
						"column": 3
					},
					"children": {}
				},
				"{functionExpression:(expression:0)}": {
					"id": 11,
					"type": "ArrowFunctionExpression",
					"identifierType": "Expression",
					"identifier": "{functionExpression:(expression:0)}",
					"beginLoc": {
						"line": 27,
						"column": 42
					},
					"endLoc": {
						"line": 29,
						"column": 3
					},
					"children": {}
				},
				"{method:asyncMethod}": {
					"id": 12,
					"type": "MethodDefinition",
					"identifierType": "Name",
					"identifier": "{method:asyncMethod}",
					"beginLoc": {
						"line": 31,
						"column": 2
					},
					"endLoc": {
						"line": 33,
						"column": 3
					},
					"children": {}
				},
				"{method:staticMethod}": {
					"id": 13,
					"type": "MethodDefinition",
					"identifierType": "Name",
					"identifier": "{method:staticMethod}",
					"beginLoc": {
						"line": 35,
						"column": 2
					},
					"endLoc": {
						"line": 37,
						"column": 3
					},
					"children": {}
				},
				"{method:method}": {
					"id": 14,
					"type": "MethodDefinition",
					"identifierType": "Name",
					"identifier": "{method:method}",
					"beginLoc": {
						"line": 47,
						"column": 2
					},
					"endLoc": {
						"line": 49,
						"column": 3
					},
					"children": {}
				},
				"{method:(expression:0)}": {
					"id": 15,
					"type": "MethodDefinition",
					"identifierType": "Expression",
					"identifier": "{method:(expression:0)}",
					"beginLoc": {
						"line": 51,
						"column": 2
					},
					"endLoc": {
						"line": 53,
						"column": 3
					},
					"children": {
						"{functionExpression:(expression:0)}": {
							"id": 16,
							"type": "ArrowFunctionExpression",
							"identifierType": "Expression",
							"identifier": "{functionExpression:(expression:0)}",
							"beginLoc": {
								"line": 51,
								"column": 4
							},
							"endLoc": {
								"line": 51,
								"column": 67
							},
							"children": {}
						}
					}
				},
				"{method:(expression:1)}": {
					"id": 17,
					"type": "MethodDefinition",
					"identifierType": "Expression",
					"identifier": "{method:(expression:1)}",
					"beginLoc": {
						"line": 55,
						"column": 2
					},
					"endLoc": {
						"line": 57,
						"column": 3
					},
					"children": {}
				}
			}
		}
	}
}